service-commands/scripts/allClockValues.js

```diff
@@ -0,0 +1,230 @@
+const fs = require("fs")
+const http = require("http")
+const https = require("https")
+
+const axios = require("axios")
+const retry = require("async-retry")
+const { map } = require("lodash")
+
+axios.defaults.timeout = 5000
+axios.defaults.httpAgent = new http.Agent({ timeout: 5000 })
+axios.defaults.httpsAgent = new https.Agent({ timeout: 5000 })
+
+function makeRequest(request) {
+  return retry(() => axios(request), { retries: 3 })
+}
+
+/**
+ * @param {string} discoveryProvider - Discovery Provider endpoint
+ * @param {number} offset
+ * @param {number} limit
+ * @returns {Array<Object>} userBatch
+ */
+async function getUsersBatch(discoveryProvider, offset, limit) {
+  return (
+    await makeRequest({
+      method: "get",
+      url: "/users",
+      baseURL: discoveryProvider,
+      params: { offset, limit, is_creator: true },
+    })
+  ).data.data.map((user) => ({
+    ...user,
+    creator_node_endpoint: user.creator_node_endpoint
+      ? user.creator_node_endpoint.split(",").filter((endpoint) => endpoint)
+      : [],
+  }))
+}
+
+/**
+ * @param {string} discoveryProvider - Discovery Provider endpoint
+ * @param {number} batchSize - Batch Size to use for each request
+ * @returns {Object} trackCids - A object mapping user id to a list of track cids
+ */
+async function getTrackCids(discoveryProvider, batchSize) {
+  const trackCids = {}
+
+  let moreTracksRemaining = true
+  for (let offset = 0; moreTracksRemaining; offset += batchSize) {
+    console.time(`Fetching tracks (${offset} - ${offset + batchSize})`)
+
+    const tracksBatch = (
+      await makeRequest({
+        method: "get",
+        url: "/tracks",
+        baseURL: discoveryProvider,
+        params: { offset, limit: batchSize },
+      })
+    ).data.data
+    moreTracksRemaining = tracksBatch.length === batchSize
```

Comment:
is this the right terminal condition for this loop?

Reply:
This is because I am not sure how to get the total number of tracks.

`tracksBatch.length` is the number of items in the iteration. In the final iteration, this is guaranteed to be less than the `batchSize`, and therefore we can terminate the loop.

Reply:
i think https://discoveryprovider3.staging.audius.co/latest/track or https://discoveryprovider3.audius.co/latest/track and production

Reply:
Nice, thanks, I'll use this instead!